### PR TITLE
 exercise46.py

### PR DESCRIPTION
**Pull request by amymhaddad**
@paulghaddad This program works, and I also tried this same program with a large file (ie, the book of Alice in Wonderland) and it also works. 

However, in the larger file there's punctuation next to many of the words, so the counter counts those as new words (ie, many. OR yes,). So I tried using a regular expression but it wasn't working (re.split(r 'W+', word) at line 24. I used this to remove the punctuation at the end of the word. Is this the best way to do it? 

Here are the instructions for this exercise:
Create a program that reads in a file and counts the frequency of words in the file. Then construct a histogram displaying the words and the frequency, and display the histogram to the screen. Given the text file words.txt with this content badger badger badger badger mushroom mushroom
       snake badger badger badger

the program would produce the following output:
       badger:   *******
       mushroom: **
       snake:    *

### CODE REVIEW

File: exercise46.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# -*- coding: utf-8 -*-
+"""
+Created on Thu May 10 15:34:22 2018
+
+@author: amyhaddad
+"""
+
+FILENAME = 'words.txt'
+string_of_words = ''
+
```
Comment (amymhaddad) [inline exercise46.py]: @paulghaddad can use list comprehenion when opening and reading a file?

File: exercise46.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+with open(FILENAME) as f_object:
+    for contents in f_object:
+        normalized_content = contents.lower().strip("\n")
+        string_of_words += normalized_content + ' '
+
+list_of_words = string_of_words.split()
+unique_words = set(list_of_words)
```
Comment (paulghaddad) [inline exercise46.py]: Wow, good thinking here to use `set`! 🥇

File: exercise46.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+final_format = []
+
+for word in unique_words:
```
Comment (paulghaddad) [inline exercise46.py]: Really nice algorithm! I like your thinking here!

File: exercise46.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+with open(FILENAME) as f_object:
+    for contents in f_object:
+        normalized_content = contents.lower().strip("\n")
+        string_of_words += normalized_content + ' '
+
+list_of_words = string_of_words.split()
+unique_words = set(list_of_words)
+
+final_format = []
```
Comment (paulghaddad) [inline exercise46.py]: better variable name...

File: exercise46.py (hunk 1 above)
Comment (paulghaddad) [reply exercise46.py]: No, you'll always want to use a look when dealing with files...that's what you normally see.

File: exercise46.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+FILENAME = 'words.txt'
+string_of_words = ''
+
+with open(FILENAME) as f_object:
+    for contents in f_object:
+        normalized_content = contents.lower().strip("\n")
+        string_of_words += normalized_content + ' '
+
+list_of_words = string_of_words.split()
```
Comment (paulghaddad) [inline exercise46.py]: I tried testing your regex but couldn't replicate it. Maybe we can sit down together...

It _should_ work as `[^a-zA-Z0-9_]` is the definition of `\W`

**Comment by amymhaddad** [comment]
@paulghaddad This file is updated! I'll open a new branch with the regex version. Let's just get this one approved :)

**Pull request closed without merging**